crates/hyperswitch_connectors/src/connectors/fiuu/transformers.rs

```diff
@@ -1,226 +1,837 @@
+                    Ok(Self {
+                        status: enums::AttemptStatus::AuthenticationPending,
+                        response: Ok(PaymentsResponseData::TransactionResponse {
+                            resource_id: ResponseId::ConnectorTransactionId(data.txn_id),
+                            redirection_data,
+                            mandate_reference: None,
+                            connector_metadata: None,
+                            network_txn_id: None,
+                            connector_response_reference_id: None,
+                            incremental_authorization_allowed: None,
+                            charge_id: None,
+                        }),
+                        ..item.data
+                    })
+                }
+
+                RequestData::NonThreeDS(non_threeds_data) => Ok(Self {
+                    status: match non_threeds_data.status.as_str() {
+                        "00" => {
+                            if item.data.request.is_auto_capture()? {
+                                Ok(enums::AttemptStatus::Charged)
+                            } else {
+                                Ok(enums::AttemptStatus::Authorized)
+                            }
+                        }
+                        "11" => Ok(enums::AttemptStatus::Failure),
+                        "22" => Ok(enums::AttemptStatus::Pending),
+                        other => Err(errors::ConnectorError::UnexpectedResponseError(
+                            bytes::Bytes::from(other.to_owned()),
+                        )),
+                    }?,
+                    response: Ok(PaymentsResponseData::TransactionResponse {
+                        resource_id: ResponseId::ConnectorTransactionId(data.txn_id),
+                        redirection_data: None,
+                        mandate_reference: None,
+                        connector_metadata: None,
+                        network_txn_id: None,
+                        connector_response_reference_id: None,
+                        incremental_authorization_allowed: None,
+                        charge_id: None,
+                    }),
+                    ..item.data
+                }),
+            },
         }
     }
 }
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
-pub enum FiuuPaymentStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct FiuuRefundRequest {
+    pub refund_type: RefundType,
+    #[serde(rename = "MerchantID")]
+    pub merchant_id: String,
+    #[serde(rename = "RefID")]
+    pub ref_id: String,
+    #[serde(rename = "TxnID")]
+    pub txn_id: String,
+    pub amount: StringMajorUnit,
+    pub signature: Secret<String>,
+}
+#[derive(Debug, Serialize, Display)]
+pub enum RefundType {
+    P,
```

Comment:
Can we preferably use a verbose name with `serde(rename = "P")` here, instead of using a single character enum variant?